### PR TITLE
test(audio,midi,runtime): expand phase3 coverage batch

### PR DESCRIPTION
**Pull request by danielraffel**
## Summary
- add audio, MIDI, state, runtime, HTTP, SIMD, sync, and parser coverage across 31 focused commits
- isolate temp-path failure cases flagged by review using unique TemporaryFile-derived paths
- rebase onto current main after #2635 so this run includes the latest CI/test hardening

## Local validation
- cmake --build build-gpu-off --target pulp-test-async-stream pulp-test-audio pulp-test-audio-edge-cases pulp-test-audio-focus pulp-test-license pulp-test-memory-message-channel pulp-test-midi pulp-test-midi-ci pulp-test-mpe-buffer pulp-test-mpe-synth-voice pulp-test-network-stream pulp-test-preset-manager pulp-test-properties pulp-test-raw-midi-parser pulp-test-running-status pulp-test-runtime-utils pulp-test-simd pulp-test-state pulp-test-sync pulp-test-ump-buffer-conversion pulp-test-xml-zip
- affected test binaries passed directly
- git diff --check origin/main...HEAD
- python3 tools/import-validation/check-source-contracts.py --strict
- python3 tools/import-validation/test_source_contracts.py
- python3 tools/scripts/skill_sync_check.py --mode=report --base origin/main
- python3 tools/scripts/version_bump_check.py --mode=report --base origin/main

### CODE REVIEW

File: test/test_runtime_utils.cpp [hunk 1]
```diff
@@ -756,9 +842,27 @@ TEST_CASE("run_process honors working directory and preserves spaced arguments",
...
 
-    std::error_code ec;
-    std::filesystem::remove(dir, ec);
+TEST_CASE("run_process rejects missing working directories",
+          "[runtime][child_process][coverage][phase3]") {
+#if defined(_WIN32) || defined(__APPLE__) || defined(__linux__)
+    const auto missing_dir =
+        (std::filesystem::temp_directory_path()
+         / "pulp-run-process-missing-working-dir").string();
+    REQUIRE_FALSE(std::filesystem::exists(missing_dir));
```
[the remark this answers is not in the document]
Comment (danielraffel) [reply test/test_runtime_utils.cpp]: Addressed in 3414fdcca: this now derives a unique marker path with TemporaryFile, removes it before calling run_process, and asserts that unique path is absent.

File: test/test_runtime_utils.cpp [hunk 2]
```diff
@@ -1489,6 +1593,62 @@ TEST_CASE("HTTP helpers round-trip against a loopback server",
...
+TEST_CASE("HTTP download reports unwritable output paths after successful fetch",
+          "[runtime][http][coverage][phase3]") {
+    const auto missing_dir_output =
+        (std::filesystem::temp_directory_path()
+         / "pulp-http-download-missing-dir"
+         / "artifact.bin").string();
+    auto exchange = serve_one_http_response("DOWNLOAD", "/unwritable.bin", missing_dir_output);
+
+    REQUIRE(exchange.accepted);
+    REQUIRE_FALSE(exchange.downloaded);
```
[the remark this answers is not in the document]
Comment (danielraffel) [reply test/test_runtime_utils.cpp]: Addressed in 3414fdcca: the unwritable download case now uses a unique missing directory derived from TemporaryFile and cleans it afterward.
